codec-http2/src/main/java/io/netty/handler/codec/http2/AbstractHttp2ConnectionHandlerBuilder.java

```diff
@@ -334,7 +335,10 @@ protected T build() {
     }
 
     private T buildFromConnection(Http2Connection connection) {
-        Http2FrameReader reader = new DefaultHttp2FrameReader(isValidateHeaders());
+        Long maxHeaderListSize = initialSettings.maxHeaderListSize();
+        Http2FrameReader reader = new DefaultHttp2FrameReader(maxHeaderListSize == null ?
```

Comment:
Nah. Unless an exception is thrown before we transfer close responsibility to `DefaultHttp2ConnectionDecoder`, or before we close on line 356. We can be more defensive about exceptions but practically I'm not sure it will make a difference.